lang/csharp/src/apache/main/CodeGen/CodeGen.cs

```diff
@@ -220,7 +227,10 @@ protected virtual SchemaNames generateNames(Schema schema)
         protected virtual void addName(Schema schema, SchemaNames names)
         {
             NamedSchema ns = schema as NamedSchema;
-            if (null != ns) if (names.Contains(ns.SchemaName)) return;
+            if (ns != null && names?.Contains(ns.SchemaName) == true)
```

Comment:
Why names`?` ?
AFAIS it cannot be `null`.
Now GitHub reports the following for line 250:
```
Check warning on line 250 in lang/csharp/src/apache/main/CodeGen/CodeGen.cs

Code scanning

Dereferenced variable may be null

Warning
Variable
may be null here as suggested by
null check.
Show more details 
```

Reply:
Any reason to add `== true` ?

Reply:
Putting names? would be the equivalent of (assuming names is null) null == true.  I've reverted the change back.  I think I was future proofing some null argument checks.